src/test/java/de/mediathekview/mserver/crawler/wdr/tasks/WdrFilmDetailTaskTest.java

```diff
@@ -0,0 +1,202 @@
+                  "",
+                  new GeoLocations[] {GeoLocations.GEO_NONE}),
+              new WdrFilmDetailTaskTestData(
+                  "http://www1.wdr.de/mediathek/video/sendungen/wdr-aktuell/video-deutschlandtrend-und-jusos-no-groko-100.html",
+                  "/wdr/wdr_film_part2.html",
+                  "/ondemand/158/1583857.js",
+                  "/wdr/wdr_film_part2.js",
+                  "/i/medp/ondemand/weltweit/fsk0/158/1583472/,1583472_18232890,1583472_18232889,1583472_18232891,1583472_18232887,1583472_18232888,.mp4.csmil/master.m3u8",
+                  "/wdr/wdr_film_part2.m3u8",
+                  "WDR aktuell",
+                  "Deutschlandtrend und Jusos No-GroKo",
+                  LocalDateTime.of(2018, 2, 16, 21, 45, 0),
+                  Duration.ofSeconds(0),
+                  "",
+                  "",
+                  "https://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/158/1583472/,1583472_18232890,1583472_18232889,1583472_18232891,1583472_18232887,1583472_18232888,.mp4.csmil/index_0_av.m3u8",
+                  "https://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/158/1583472/,1583472_18232890,1583472_18232889,1583472_18232891,1583472_18232887,1583472_18232888,.mp4.csmil/index_2_av.m3u8",
+                  "",
+                  new GeoLocations[] {GeoLocations.GEO_NONE})
+            }
+          }
+        });
+  }
+
+  private final WdrFilmDetailTaskTestData[] expectedFilms;
+
+  public WdrFilmDetailTaskTest(WdrFilmDetailTaskTestData[] aExpectedFilms) {
+    expectedFilms = aExpectedFilms;
+  }
+
+  @Test
+  public void test() throws IOException {
+    setupHeadRequestForFileSize();
+
+    Map<String, String> urlMapping = new HashMap<>();
+    for (WdrFilmDetailTaskTestData expected : expectedFilms) {
+      urlMapping.put(expected.getRequestUrl(), expected.getFilmPageFile());
+      setupSuccessfulResponse(expected.getJsUrl(), expected.getJsFile());
+      setupSuccessfulResponse(expected.getM3u8Url(), expected.getM3u8File());
+    }
+    JsoupMock.mock(urlMapping);
+
+    final String topic = expectedFilms[0].getTopic();
+    final String requestUrl = expectedFilms[0].getRequestUrl();
+
+    final Set<Film> actual =
+        new WdrFilmDetailTask(createCrawler(), createCrawlerUrlDto(topic, requestUrl)).invoke();
+
+    assertThat(actual, notNullValue());
+    assertThat(actual.size(), equalTo(expectedFilms.length));
+
+    final Object[] actualArray = actual.toArray();
+    for (int i = 0; i < actual.size(); i++) {
+      Film actualFilm = (Film) actualArray[i];
+
+      Optional<WdrFilmDetailTaskTestData> expectedFilmOptional = getTestData(actualFilm.getTitel());
+      assertThat(expectedFilmOptional.isPresent(), equalTo(true));
+
+      WdrFilmDetailTaskTestData expectedFilm = expectedFilmOptional.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "expectedFilmOptional.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)